packages/editor-ui/src/views/NodeView.vue

```diff
@@ -3771,21 +3779,36 @@ export default defineComponent({
 			nodeTypes: Array<{ nodeTypeName: string; position: XYPosition }>,
 			dragAndDrop: boolean,
 		) {
-			nodeTypes.forEach(({ nodeTypeName, position }, index) => {
-				const isManualTrigger = nodeTypeName === MANUAL_TRIGGER_NODE_TYPE;
-				const openNDV = !isManualTrigger && (nodeTypes.length === 1 || index > 0);
+			const enrichedNodeTypes = nodeTypes
+				.map((nodeType): Array<{ nodeTypeName: string; position: XYPosition; name?: string }> => {
+					if (nodeType.nodeTypeName === SPLIT_IN_BATCHES_NODE_TYPE) {
+						return [
+							nodeType,
+							{ nodeTypeName: NO_OP_NODE_TYPE, position: nodeType.position, name: 'Replace Me' },
+						];
+					}
+					return [nodeType];
+				})
+				.flat();
+
+			const openNDVForIndex = enrichedNodeTypes.findIndex(
+				({ nodeTypeName }) => nodeTypeName !== MANUAL_TRIGGER_NODE_TYPE,
+			);
+
+			enrichedNodeTypes.forEach(({ nodeTypeName, position, name }, index) => {
 				void this.addNode(
 					nodeTypeName,
-					{ position, dragAndDrop },
-					openNDV,
+					{ position, dragAndDrop, name },
+					openNDVForIndex === index,
 					true,
-					nodeTypes.length > 1 && index < 1,
+					index !== 0,
 				);
+
 				if (index === 0) return;
 				// If there's more than one node, we want to connect them
 				// this has to be done in mutation subscriber to make sure both nodes already
 				// exist
-				const actionWatcher = this.workflowsStore.$onAction(({ name, after, args }) => {
+				const unsubscribeActionWatcher = this.workflowsStore.$onAction(({ name, after, args }) => {
```

Comment:
not a fan of this pattern.. since this happens after the node is injected and very custom solution for one node.. ideally the node creator should emit an event that includes all three nodes and how they are connected (like a pasted workflow).. 
might require a bit of refactoring but would be much cleaner and more flexible.. I believe that's how we should have implemented the manual node trigger with actions.. 
let me know if this is not clear.. might help for us to sync on this